mesh_tools/create_SCRIP_files/create_SCRIP_file_from_planar_rectangular_grid.py

```diff
@@ -0,0 +1,172 @@
+grid_dims = fout.createVariable('grid_dims', 'i4', ('grid_rank',))
+
+
+# Create matrices of x,y
+print ('Building matrix version of x, y locations.')
+xmatrix, ymatrix = np.meshgrid(x, y)
+xc = np.append(x[:] - dx/2.0, x[-1] + dx/2.0 )  # get a copy of x that is on the staggered grid and includes both bounding edges
+yc = np.append(y[:] - dy/2.0, y[-1] + dy/2.0 )  # get a copy of y that is on the staggered grid and includes both bounding edges
+xcmatrix, ycmatrix = np.meshgrid(xc, yc)
+
+# Unproject to lat/long for grid centers and grid corners
+print ('Unprojecting.')
+
+xmatrix_flat = xmatrix.flatten(order='C')  # Flatten using C indexing
+ymatrix_flat = ymatrix.flatten(order='C')
+
+# make a CRS (coordinate reference system) for projections from the Proj string:
+crs_in = CRS.from_proj4(projections[options.projection])
+crs_out = CRS.from_proj4(projections['latlon'])
+
+# building a transformer
+t = Transformer.from_crs(crs_in,crs_out)
+
+# transform the original grid into the lat-lon grid
+grid_center_lon[:], grid_center_lat[:] = t.transform(xmatrix_flat, ymatrix_flat)
+
+
+# Now fill in the corners in the right locations
+stag_lon, stag_lat = t.transform(xcmatrix, ycmatrix)
+print ('Filling in corners of each cell.')
+grid_corner_lon_local = np.zeros( (nx * ny, 4) )  # It is WAYYY faster to fill in the array entry-by-entry in memory than to disk.
+grid_corner_lat_local = np.zeros( (nx * ny, 4) )
+for j in range(ny):
+  for i in range(nx):
+    iCell = j*nx + i
+
+    grid_corner_lon_local[iCell, 0] = stag_lon[j, i]
+    grid_corner_lon_local[iCell, 1] = stag_lon[j, i+1]
+    grid_corner_lon_local[iCell, 2] = stag_lon[j+1, i+1]
+    grid_corner_lon_local[iCell, 3] = stag_lon[j+1, i]
+    grid_corner_lat_local[iCell, 0] = stag_lat[j, i]
+    grid_corner_lat_local[iCell, 1] = stag_lat[j, i+1]
+    grid_corner_lat_local[iCell, 2] = stag_lat[j+1, i+1]
+    grid_corner_lat_local[iCell, 3] = stag_lat[j+1, i]
+
+grid_corner_lon[:] = grid_corner_lon_local[:]
+grid_corner_lat[:] = grid_corner_lat_local[:]
+
+grid_imask[:] = 1  # For now, assume we don't want to mask anything out - but eventually may want to exclude certain cells from the input mesh during interpolation
+
+# set the grid dimension based on the grid rank
+if int(options.gridRank) == 1:
+    grid_dims[:] = (nx * ny)
+elif int(options.gridRank) == 2:
+    grid_dims[:] = [nx , ny]
+
+
+if options.plot:
+    print("plotting is on")
+    # plot some stuff
+    #plot a single point
+    i=-1
+    plt.plot(grid_center_lon[i], grid_center_lat[i], 'o')
+    plt.plot(grid_corner_lon[i, 0], grid_corner_lat[i, 0], 'kx')
+    plt.plot(grid_corner_lon[i, 1], grid_corner_lat[i, 1], 'bx')
+    plt.plot(grid_corner_lon[i, 2], grid_corner_lat[i, 2], 'cx')
+    plt.plot(grid_corner_lon[i, 3], grid_corner_lat[i, 3], 'gx')
+    plt.show()
+
+    #plot all points
+    plt.plot(grid_center_lon[:], grid_center_lat[:], 'bo')
+    plt.plot(grid_corner_lon[:], grid_corner_lat[:], 'g.')
+    plt.show()
```

Comment:
This is super picky since this is just for debugging but maybe plot the two figures and then show them both:
```suggestion
    plt.figure(1)
    plt.plot(grid_center_lon[i], grid_center_lat[i], 'o')
    plt.plot(grid_corner_lon[i, 0], grid_corner_lat[i, 0], 'kx')
    plt.plot(grid_corner_lon[i, 1], grid_corner_lat[i, 1], 'bx')
    plt.plot(grid_corner_lon[i, 2], grid_corner_lat[i, 2], 'cx')
    plt.plot(grid_corner_lon[i, 3], grid_corner_lat[i, 3], 'gx')

    #plot all points
    plt.figure(2)
    plt.plot(grid_center_lon[:], grid_center_lat[:], 'bo')
    plt.plot(grid_corner_lon[:], grid_corner_lat[:], 'g.')
    plt.show()
```